rotkehlchen/api/v1/resources.py

```diff
@@ -2319,28 +2327,108 @@ class LocationAssetMappingsResource(BaseMethodView):
 
     @use_kwargs(post_schema, location='json')
     def post(self, filter_query: LocationAssetMappingsFilterQuery) -> Response:
-        return self.rest_api.query_location_asset_mappings(filter_query=filter_query)
+        def location_mapping_reader(entry: dict[str, Any]) -> dict[str, Any]:
```

Comment:
can you move this out of the function? Having it as an inner function it always get redefined every time this function is called.